### PR TITLE
bpo-35728: Add root parameter to tkinter.font.nametofont()

### PR DESCRIPTION
**Pull request by desmondcheongzx**
Add a root parameter to tkinter.font.nametofont() so that the function can be called even when default root is disabled.
https://bugs.python.org/issue35728

https://bugs.python.org/issue35728

### CODE REVIEW

File: Doc/library/tkinter.font.rst [hunk 1]
```diff
@@ -91,6 +91,6 @@ The different font weights and slants are:
 
    Return the names of defined fonts.
 
-.. function:: nametofont(name)
+.. function:: nametofont(name, root=None)
```
Comment (serhiy-storchaka) [inline Doc/library/tkinter.font.rst:94]: Please add a `versionchanged` directive.

**Review by serhiy-storchaka — APPROVED** [review]
LGTM.

File: Doc/library/tkinter.font.rst (hunk 1 above)
Comment (terryjreedy) [reply Doc/library/tkinter.font.rst:94]: Done

**Review by terryjreedy — APPROVED** [review]
The function is not terribly useful, but now is consistent with other functions.  Tests look good.  Thanks..

File: Doc/library/tkinter.font.rst [hunk 2]
```diff
@@ -91,6 +91,9 @@ The different font weights and slants are:
...
 
-   Return a :class:`Font` representation of a tk named font.
\ No newline at end of file
+   Return a :class:`Font` representation of a tk named font.
+
+  .. versionchanged:: 3.10
```
Comment (Fidget-Spinner) [inline Doc/library/tkinter.font.rst]: Seems to be missing a space here, indentation in rST is 3 spaces. On some versions of sphinx this causes visual differences.

```suggestion
   .. versionchanged:: 3.10
      Added the *root* keyword argument.   
```
Other than that, LGTM :).

Comment (desmondcheongzx) [reply Doc/library/tkinter.font.rst]: Thanks for catching that! I've updated the documentation accordingly

**Review by Fidget-Spinner — APPROVED** [review]

**Comment by terryjreedy** [comment]
Both Travis and Pipelines failed the doc check of the news item, which I believe is unchanged from the previous run that passed.  I don't understand the complaint or know if the test was changed.  I restarted Travis and now it passes.  Maybe the transient failure was in github.  I don't see any way to rerun Pipelines, but since it is the same test, and it passes before, I would consider ignoring it.
```
python3 tools/rstlint.py ../Misc/NEWS.d/next/
[2] ../Misc/NEWS.d/next/Library/2020-12-23-19-43-06.bpo-42727.WH3ODh.rst:1: default role used
[2] ../Misc/NEWS.d/next/Library/2020-12-23-19-43-06.bpo-42727.WH3ODh.rst:2: default role used
2 problems with severity 2 found.
Makefile:204: recipe for target 'check' failed
```

**Comment by Fidget-Spinner** [comment]
> Both Travis and Pipelines failed the doc check of the news item, which I believe is unchanged from the previous run that passed. I don't understand the complaint or know if the test was changed. I restarted Travis and now it passes. Maybe the transient failure was in github. I don't see any way to rerun Pipelines, but since it is the same test, and it passes before, I would consider ignoring it.

The error means that another news item in another bpo used single backticks rather than double backticks for styling code.  It was recently fixed in 8badadec53cbf9dc049c5b54198c5689481e3f3f. Updating this PR to the latest master branch should be able to fix it.

**Comment by terryjreedy** [comment]
I just saw the checkin of #23932 which changed the news entry for #23917, which at least once had the exact same failure, by doubling the backticks.  Perhaps that is needed here also.

**Comment by terryjreedy** [comment]
Messages crossed.  @desmondcheongzx please do the update with fetch and merge (or simple pull) rather than with 'force' argument, as some mistakenly do.  It is too easy to mess up PR with the latter.

**Comment by desmondcheongzx** [comment]
Got it, I've made the appropriate update.

**Comment by serhiy-storchaka** [comment]
Thank you for your contribution @desmondcheongzx!

**Comment by desmondcheongzx** [comment]
Thank you for the reviews!

**Pull request merged**